app/controllers/search_controller.rb

```diff
@@ -0,0 +1,10 @@
+class SearchController < ApplicationController
+
+  def index
+    @nearest_elec_station = StationFacade.get_nearest_station(params[:location])
+    start_address = params[:location]
+    end_address = @nearest_elec_station.first.full_address
+    @directions_information = DirectionFacade.get_directions_info(start_address, end_address)
+  end
```

Comment:
This all looks good. One thing that you can do is call to two different services in one method. For example, currently you are getting the station and then you are getting directions. Instead, you could call to one method that gets the nearest station, and directions for that nearest station, and returns everything in one object rather than two.